File: pubsub/src/main/java/glitch/pubsub/events/SubscriptionEvent.java
package glitch.pubsub.events;

import glitch.api.ws.events.IEvent;
import glitch.pubsub.GlitchPubSub;
import glitch.pubsub.Topic;
import glitch.pubsub.object.json.SubscriptionMessage;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class SubscriptionEvent extends AbstractMessageEvent<SubscriptionMessage> implements IEvent<GlitchPubSub> {
    public SubscriptionEvent(GlitchPubSub client, Topic topic, SubscriptionMessage data) {
        super(client, topic, data);
    }
}

File: pubsub/src/main/kotlin/glitch/pubsub/events/SubscriptionEvent.kt
package glitch.pubsub.events

import glitch.api.ws.events.IEvent
import glitch.pubsub.GlitchPubSub
import glitch.pubsub.Topic
import glitch.pubsub.`object`.json.SubscriptionMessage

class SubscriptionEvent(client: GlitchPubSub, topic: Topic, data: SubscriptionMessage) : AbstractMessageEvent<SubscriptionMessage>(client, topic, data), IEvent<GlitchPubSub> {

}
